api/notd/manager.py

```diff
@@ -337,6 +340,40 @@ async def list_account_tokens(self, accountAddress: str, limit: int, offset: int
         sortedTokenOwnershipTuples = sorted(tokenOwnershipTuples, key=lambda tuple: tuple[2], reverse=True)
         return [Token(registryAddress=registryAddress, tokenId=tokenId) for (registryAddress, tokenId, _) in sortedTokenOwnershipTuples]
 
+    async def list_account_tokens_with_delegate_tokens(self, accountAddress: str, limit: int, offset: int) -> List[AccountToken]:
+        delegations = await self.delegationManager.get_delegations(delegateAddress=accountAddress)
+        accountTokens = []
+        for delegation in delegations:
+            tokens = await self.list_account_tokens(accountAddress=delegation.vaultAddress, limit=limit, offset=offset)
+            if delegation.delegationType == "ALL":
+                accountTokens += [AccountToken(registryAddress=token.registryAddress, tokenId=token.tokenId, ownerAddress=delegation.vaultAddress) for token in tokens]
+            if delegation.delegationType == "CONTRACT":
+                accountTokens += [AccountToken(registryAddress=token.registryAddress, tokenId=token.tokenId, ownerAddress=delegation.vaultAddress) for token in tokens if delegation.contractAddress == token.registryAddress]
+            if delegation.delegationType == "TOKEN":
+                #NOTE(Femi-Ogunkola): registryAddress=delegation.contractAddress, tokenId=delegation.tokenId but typing error with Optional[str]
+                accountTokens += [AccountToken(registryAddress=token.registryAddress, tokenId=token.tokenId, ownerAddress=delegation.vaultAddress) for token in tokens if delegation.contractAddress == token.registryAddress and delegation.tokenId == token.tokenId]
+
+        tokenSingleOwnerships = await self.retriever.list_token_ownerships(
+            fieldFilters=[
+                StringFieldFilter(fieldName=TokenOwnershipsTable.c.ownerAddress.key, eq=accountAddress),
+            ],
+            orders=[Order(fieldName=TokenOwnershipsTable.c.transferDate.key, direction=Direction.DESCENDING)],
+            limit=limit+offset,
```

Comment:
this limit+offset cannot work here because you have two separate queries. you should maybe try to use the view here instead